akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorContextAdapter.scala

```diff
@@ -46,6 +46,12 @@ import scala.concurrent.duration._
             // child that was already stopped
           }
       }
+    } else if (self == child) {
+      throw new IllegalArgumentException(
+        "Only direct children of an actor can be stopped through the actor context, " +
+          s"but you try to stop [$self] by passed its ActorRef to the `stop` method. " +
+          "Stopping self has to be expressed as explicitly returning a Stop Behavior " +
+          "with `Behaviors.stopped`.")
```

Comment:
"but you tried to stop [$self] by passing its ActorRef to the"

Reply:
@johanandren thanks, done.